spec/errors.md

```diff
@@ -302,11 +324,9 @@ An **_<dfn>Unsupported Statement</dfn>_** error occurs when a message includes a
 > * {{The value is not one.}}
 > ```
 
-## Formatting Errors
+### Formatting Error
 
-**_<dfn>Formatting Errors</dfn>_** occur during the formatting of a resolved value,
-for example when encountering a value with an unsupported type
-or an internally inconsistent set of options.
+A **_<dfn>Formatting Error</dfn>_** occurs during the formatting of a resolved value.
 
 > For example, attempting to format any of the following messages
 > might result in a _Formatting Error_ if done within a context that
```

Comment:
> ```
> Hello, {horse :get field=name}!
> ```

Wouldn't this be a Bad Operand error, because "_`:get`… requires its argument to be an object_"?

> ```
> Hello, {$user :get}!
> ```

Wouldn't this be a Bad Option error, because `:get` requires "_an option field to be provided with a string value_"?

> ```
> .local $id = {$user :get field=id}
> {{Hello, {$id :get field=name}!}}
> ```

Wouldn't this be a Bad Operand error, because "_`:get`… requires its argument to be an object_"?

> ```
> Your {$field} is {$id :get field=$field}
> ```

Wouldn't this be an Unresolved Variable error, because `$id` cannot be resolved?

----

As described in [Formatting](https://github.com/unicode-org/message-format-wg/blob/b76dc21172364da7cf7f30ea5ed91e4e7137b9d4/spec/formatting.md#formatting) (**bold emphasis** mine),
> After <em>pattern selection</em>, each <em>text</em> and <em>placeholder</em> part of the selected <em>pattern</em> is **resolved** and formatted.

Formatting is separated from formatting context construction pattern selection, but all those phases include resolution and therefore the potential for resolution errors—in other words, "Resolution Errors" as a category does not seem to imply specificity to any particular phase. If the goal is to establish a _tree_ of error types rather than merely a DAG, then I think it is necessary to either include phase in the name of errors that could otherwise occur in more than one or to define that e.g. other error types are subsumed into Formatting Error during the formatting phase.

To be frank, this all seems rather messy. We should probably just drop Selection Error and Formatting Error and instead recommend that implementations communicate in another way the point at which error-free formatting was interrupted, allowing the errors theirselves to maintain the desired level of specificity.